include/rmm/rmm.hpp

```diff
@@ -128,14 +129,14 @@ inline rmmError_t alloc(T** ptr, size_t size, cudaStream_t stream, const char* f
   }
 
   if (!ptr) return RMM_ERROR_INVALID_ARGUMENT;
+  try{
```

Comment:
The only above safety check that needs to be preserved is:
```
 if (!ptr && !size) {
    return RMM_SUCCESS;
}
```
All the others can be deleted as they'll be handled by the memory resource.

Reply:
ok so now it looks like 
`
  if (!ptr && !size) {
    return RMM_SUCCESS;
  } 
  try{
     *ptr = static_cast<T*>(
      rmm::mr::get_default_resource()->allocate(size,stream));

  }catch(std::exception e){
    *ptr = nullptr;
    return RMM_ERROR_OUT_OF_MEMORY;
  }
`

Reply:
i think we need this check   if (!ptr) return RMM_ERROR_INVALID_ARGUMENT;
else the test that sends it in a nullptr crashes when we derefrence *ptr

Reply:
Yes, good point.